spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/metrics/JvmMetricsAutoConfigurationTests.java

```diff
@@ -43,41 +49,53 @@
-						.hasSingleBean(JvmThreadMetrics.class).hasSingleBean(ClassLoaderMetrics.class));
+		this.contextRunner.run((context) -> validateBaseJvmMetricsBeansArePresent());
 	}
 
 	@Test
 	void allowsCustomJvmGcMetricsToBeUsed() {
-		this.contextRunner.withUserConfiguration(CustomJvmGcMetricsConfiguration.class)
-				.run((context) -> assertThat(context).hasSingleBean(JvmGcMetrics.class).hasBean("customJvmGcMetrics")
-						.hasSingleBean(JvmMemoryMetrics.class).hasSingleBean(JvmThreadMetrics.class)
-						.hasSingleBean(ClassLoaderMetrics.class));
+		this.contextRunner.withUserConfiguration(CustomJvmGcMetricsConfiguration.class).run((context) -> {
+			validateBaseJvmMetricsBeansArePresent();
+			assertThat(context).hasBean("customJvmGcMetrics");
+		});
 	}
 
 	@Test
 	void allowsCustomJvmMemoryMetricsToBeUsed() {
-		this.contextRunner.withUserConfiguration(CustomJvmMemoryMetricsConfiguration.class)
-				.run((context) -> assertThat(context).hasSingleBean(JvmGcMetrics.class)
-						.hasSingleBean(JvmMemoryMetrics.class).hasBean("customJvmMemoryMetrics")
-						.hasSingleBean(JvmThreadMetrics.class).hasSingleBean(ClassLoaderMetrics.class));
+		this.contextRunner.withUserConfiguration(CustomJvmMemoryMetricsConfiguration.class).run((context) -> {
+			validateBaseJvmMetricsBeansArePresent();
+			assertThat(context).hasBean("customJvmMemoryMetrics");
+		});
 	}
 
 	@Test
 	void allowsCustomJvmThreadMetricsToBeUsed() {
-		this.contextRunner.withUserConfiguration(CustomJvmThreadMetricsConfiguration.class)
-				.run((context) -> assertThat(context).hasSingleBean(JvmGcMetrics.class)
-						.hasSingleBean(JvmMemoryMetrics.class).hasSingleBean(JvmThreadMetrics.class)
-						.hasSingleBean(ClassLoaderMetrics.class).hasBean("customJvmThreadMetrics"));
+		this.contextRunner.withUserConfiguration(CustomJvmThreadMetricsConfiguration.class).run((context) -> {
+			validateBaseJvmMetricsBeansArePresent();
+			assertThat(context).hasBean("customJvmThreadMetrics");
+		});
 	}
 
 	@Test
 	void allowsCustomClassLoaderMetricsToBeUsed() {
-		this.contextRunner.withUserConfiguration(CustomClassLoaderMetricsConfiguration.class)
-				.run((context) -> assertThat(context).hasSingleBean(JvmGcMetrics.class)
-						.hasSingleBean(JvmMemoryMetrics.class).hasSingleBean(JvmThreadMetrics.class)
-						.hasSingleBean(ClassLoaderMetrics.class).hasBean("customClassLoaderMetrics"));
+		this.contextRunner.withUserConfiguration(CustomClassLoaderMetricsConfiguration.class).run((context) -> {
+			validateBaseJvmMetricsBeansArePresent();
+			assertThat(context).hasBean("customClassLoaderMetrics");
+		});
+	}
+
+	@Test
+	void allowsCustomDiskSpaceMetricsToBeUsed() {
+		this.contextRunner.withUserConfiguration(CustomDiskSpaceMetricsConfiguration.class).run((context) -> {
+			validateBaseJvmMetricsBeansArePresent();
+			assertThat(context).hasBean("customDiskSpaceMetrics");
+		});
+	}
+
+	private ContextConsumer<AssertableApplicationContext> validateBaseJvmMetricsBeansArePresent() {
```

Comment:
**[UNRELATED]**: Just refactored to make it a bit more clear as to what was changing/being verified in each test.